compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -1407,11 +1407,15 @@ trait Implicits { self: Typer =>
         }
       if (ctx.reporter.hasErrors) {
         ctx.reporter.removeBufferedMessages
-        SearchFailure {
-          adapted.tpe match {
-            case _: SearchFailureType => adapted
-            case _ => adapted.withType(new MismatchedImplicit(ref, pt, argument))
-          }
+        adapted.tpe match {
+          case _: SearchFailureType => SearchFailure(adapted)
+          case _ =>
+            // Special case for `$conforms[Nothing]` and `<:<.refl[Nothing]`. Showing them
+            // up to the users brings no value, so we instead report a `NoMatchingImplicitsFailure`
+            if (adapted.show == "$conforms[Nothing]" || adapted.show == "<:<.refl[Nothing]")
```

Comment:
In case it is not recommended to use `show` to compare trees, could you please advise on how to build the following tree?

~~~
TypeApply(Ident($conforms),List(TypeTree[TypeVar(TypeParamRef(A) -> TypeRef(ThisType(TypeRef(NoPrefix,module class scala)),class Nothing))]))
~~~

In particular, I’m not sure how to construct the part of the tree that are tostringed to “module class scala” and “class Nothing”.